### DISCUSSION TITLE
quarto render: [WARNING] Could not fetch resource _template_files/...

### DISCUSSION THREAD

**Discussion by bryanwhiting** [quarto-dev/quarto-cli#4041] [Q&A]
Quen I run "quarto render" with relative paths, I get this error: 

```
poetry run quarto render \
    ../quarto/_template.qmd 
```

```
[WARNING] Could not fetch resource _template_files/libs/clipboard/clipboard.min.js
[WARNING] Could not fetch resource _template_files/libs/quarto-html/quarto.js
[WARNING] Could not fetch resource _template_files/libs/quarto-html/popper.min.js
[WARNING] Could not fetch resource _template_files/libs/quarto-html/tippy.umd.min.js
[WARNING] Could not fetch resource _template_files/libs/quarto-html/anchor.min.js
[WARNING] Could not fetch resource _template_files/libs/quarto-html/tippy.css
[WARNING] Could not fetch resource _template_files/libs/quarto-html/quarto-syntax-highlighting.css
[WARNING] Could not fetch resource _template_files/libs/bootstrap/bootstrap.min.js
[WARNING] Could not fetch resource _template_files/libs/bootstrap/bootstrap-icons.css
[WARNING] Could not fetch resource _template_files/libs/bootstrap/bootstrap.min.css
[WARNING] Could not fetch resource _template_files/libs/quarto-html/zenscroll-min.js
```
and no fancy quarto formatting appears.

But when I run it as a relative path: 
```
quarto render \
    quarto/_template.qmd
```
I don't have the same error.

**Comment by dragonstyle** [comment]
I'm not 100% sure what is happening though I'm guessing something about the first configuration is foiling our assumptions about working directories. Do you have an example repo / zip that one could use to see / reproduce this?

**Comment by asadow** [comment]
I get the same warnings when I use `quarto_render()` with something like `input = qmd/blah.qmd` and `output_file= "not-blah.qmd"`.

Interestingly the warnings and lack of formatting don't show up if I simply move blah.qmd to the project root folder and set input to match.

**Reply by cderv** [reply]
Without any example repo / zip project to look into, it will be hard to know what is happening.

I can't reproduce with simple documents and projects, so probably coming from a set of configuration or a project organisation.

Unless it is no more an issue with latest 1.3 - does this happens to you with latest 1.3 version ? https://quarto.org/docs/download/

**Reply by p-gw** [reply]
I recently also ran into this problem and tried to create a minimal project to reproduce the error: [project.zip](https://github.com/quarto-dev/quarto-cli/files/11614422/project.zip)

When rendering the document using `quarto render report/report.qmd -o alternate_name.html` I observed the following: 

- If `self-contained: false` the libs are not recognized by the final document
- If `self-contained: true` the above *could not fetch resource* warning comes up
- It only happens if the `qmd` file is in a subdirectory of the project, top-level documents seem fine

```
$ quarto --version
1.3.353
```

**Reply by mcanouil** [reply]
First `self-contained` was deprecated a while ago as described in Pandoc and Quarto documentation, `embed-resources` is the option you should use[^1].

Could you share your reproducible example either as a Git repository or as plain text?

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

[^1]: This should not change anything but avoid issues in the future.

**Reply by cderv** [reply]
> tried to create a minimal project to reproduce the error: [project.zip](https://github.com/quarto-dev/quarto-cli/files/11614422/project.zip)

@mcanouil there is project.zip linked. 

>When rendering the document using quarto render report/report.qmd -o alternate_name.html I observed the following:
>
>If self-contained: false the libs are not recognized by the final document
>If self-contained: true the above could not fetch resource warning comes up
>It only happens if the qmd file is in a subdirectory of the project, top-level documents seem fine

@p-gw I can reproduce this. I believe this is because of `-o alternate_name.html` as you are changing the output destination while it is expected to be in `_output/report` 

If you just want to change the name of the output, you should use `output-file: alternate_name` in YAML and the file will be put in the right folder. 

However, I believe we should try to handle `-o` differently or at least warn / error helpfully

I created https://github.com/quarto-dev/quarto-cli/issues/5765 to follow

**Reply by p-gw** [reply]
Thanks @cderv, not sure whether to post my answer here or in #5765.

Using the `output-file` options works as expected. 

However, the reason I posted this example is that the document name needs to be dynamic and set based on a parameter. So my real world use case is more like this example: 

````qmd
---
title: A test document
format:
    html: 
        embed-resources: true
params:
    x: "value"
---

```{r}
params$x
```
````

```sh
for value in 1 2 3
do
    quarto render report/report.qmd -P x:$value -o report_$value.html
done
```

As far as I am aware `output-file` does not support this, e.g. ``output-file: `r paste0("report_", params$x)` `` will throw an error.

**Reply by cderv** [reply]
> not sure whether to post my answer here or in https://github.com/quarto-dev/quarto-cli/issues/5765.

It would be best in #5765 to explain the context of your use case. 

> As far as I am aware output-file does not support this, e.g. output-file: `r paste0("report_", params$x)`  will throw an error.

You are trying inline knitr R code here. This won't work with Quarto document YAML. 

Possibly shortcode could be used (https://quarto.org/docs/authoring/variables.html) but I don't know if using variables would work with `output-file` field. This worth a try.
